File: MenstrualApp.py
print("MENSTRUAL CYCLE APP")

flowstart = int(input("Enter flow start day: "))

cyclelength = int(input("Enter cycle length: "))

flowduration = int(input("Enter flow duration: "))


flowend = flowstart + flowduration - 1

nextperiod = flowstart + cyclelength

ovulationday = nextperiod - 14

fertilestart = ovulationday - 5
fertileend = ovulationday + 1

safestart1 = flowend + 1
safeend1 = fertilestart - 1

safestart2 = fertileend + 1
safeend2 = nextperiod - 1


print("\nRESULTS")

print("Flow Period: Day", flowstart, "to Day", flowend)

print("Next Period Day:", nextperiod)

print("Ovulation Day:", ovulationday)

print("Fertile Period: Day", fertilestart, "to Day", fertileend)

print("Safe Period 1: Day", safestart1, "to Day", safeend1)

print("Safe Period 2: Day", safestart2, "to Day", safeend2)

File: MenstrualApp.java
import java.util.Scanner;

public class MenstrualApp {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        System.out.println(" MENSTRUAL CYCLE APP ");

        
        System.out.print("Enter flow start day: ");
        int flowStart = input.nextInt();

        
        System.out.print("Enter cycle length: ");
        int cycleLength = input.nextInt();

        
        System.out.print("Enter flow duration: ");
        int flowDuration = input.nextInt();



        
        int flowEnd = flowStart + flowDuration - 1;

        
        int nextPeriod = flowStart + cycleLength;

        
        int ovulationDay = nextPeriod - 14;

        
        int fertileStart = ovulationDay - 5;
        int fertileEnd = ovulationDay + 1;

        
        int safeStart1 = flowEnd + 1;
        int safeEnd1 = fertileStart - 1;

        
        int safeStart2 = fertileEnd + 1;
        int safeEnd2 = nextPeriod - 1;



        
        System.out.println("\n RESULTS ");

        System.out.println("Flow Period: Day " + flowStart + " to Day " + flowEnd);

        System.out.println("Next Period Day: " + nextPeriod);

        System.out.println("Ovulation Day: " + ovulationDay);

        System.out.println("Fertile Period: Day " + fertileStart + " to Day " + fertileEnd);

        System.out.println("Safe Period 1: Day " + safeStart1 + " to Day " + safeEnd1);

        System.out.println("Safe Period 2: Day " + safeStart2 + " to Day " + safeEnd2);

    }
}